auth/auth0/src/loggedInUser.graphql

```diff
@@ -0,0 +1,8 @@
+type LoggedInUserPayload {
+  id: ID!
+}
+
+extend type Query {
+  # return user data if request contains valid authentication token
+  loggedInUser: LoggedInUserPayload
```

Comment:
Is there any reason that this cannot return a `User` rather than a `LoggedInUserPayload` that just contains the user's id?  I would like to query for more user fields after logging in, but the way it is I need to make multiple queries.

Reply:
Yes there is. Resolver functions do not support returning @model Types at the moment.

Reply:
But I think that you can avoid use this method if you add some basic permissions rules

Example: https://github.com/fabien0102/climbing-app/blob/fdc17ca70336863cf68dd5f3a46b0f7076e5f31e/graphcool/graphcool.yml#L30

After I just a classic user query https://github.com/fabien0102/climbing-app/blob/master/app/queries/userQuery.graphql and it's return the user if I'm connected or null if I'm not :wink:

Reply:
@fabien0102 No, the `User` query requires you to pass in a userId. It does not support taking the user out of the token. You need a resolver function for that. You are using the `user` query, which is the current out of the box implementation, which only works for the `User` Type, and according to the documentation should be restricted to legacy projects. All authentication templates use this resolver function as a replacement.

Reply:
With header:
![image](https://user-images.githubusercontent.com/1761469/32891379-9fcb5f86-cad2-11e7-9cb4-65239f81b174.png)

Without header:
![image](https://user-images.githubusercontent.com/1761469/32891410-b8d400fa-cad2-11e7-9565-f2a5eab8f443.png)

I don't have your point ^^ (screenshot from playground on my climbing-app repo)

Reply:
I know, that's why I said: _should be_ restricted to legacy projects. Currently, it's still available on CLI projects too. But the templates take this future change into account by providing their own `loggedinUser` function.

Reply:
Too bad, it's just great like that ^^ Please don't remove this feature before be able to return the user model into `loggedInUser` 😄